test/string-class-test.js

```diff
@@ -0,0 +1,175 @@
+      expect('?'.isQuestion()).to.be.false;
+      expect('???'.isQuestion()).to.be.false;
+      expect(' ?'.isQuestion()).to.be.false;
+    });
+  });
+
+  describe('word', () => {
+    it('should return a list of words', () => {
+      expect('This is Andela'.word()).to.deep.equal(['This', 'is', 'Andela']);
+    });
+    it('should return a list of words without special characters', () => {
+      expect('$T$$%his i^s A)()ndel###a'.word()).to.deep.equal(['This', 'is', 'Andela']);
+    });
+  });
+
+  describe('wordCount', () => {
+    it('should return the correct number of words', () => {
+      expect('This is Andela TIA'.wordCount()).to.equal(4);
+      expect('$this: is[] Andel!a'.wordCount()).to.equal(3);
+    });
+  });
+
+  describe('toCurrency', () => {
+    it('should return a currency representation of a number', () => {
+      expect('10000.00'.toCurrency()).to.equal('10,000.00');
+    });
+    it('should return a currency representation with decimals', () => {
+      expect('5000'.toCurrency()).to.equal('5,000.00');
+    });
+    it('should return a currency representation with two decimals', () => {
+      expect('123456.78945'.toCurrency()).to.equal('123,456.79');
+    });
+  });
+
+  describe('fromCurrency', () => {
+    it('should return a number representation of the currency', () => {
+      expect('123,456,789.98'.fromCurrency()).to.equal('123456789.98');
+    });
+    it('should return a number representation of the currency without insignificant zeroes', () => {
+      expect('123,456,789.00'.fromCurrency()).to.equal('123456789');
+    });
+  });
+
+  describe('inverseCase', () => {
+    it('should return each letter of a string in its inverse case', () => {
+      expect('Mr. Ben'.inverseCase()).to.equal('mR. bEN');
+    });
+  });
+
+  describe('alternatingCase', () => {
+    it('should return an alternating case for each letter in a string starting with lower case', () => {
+      expect('Checkpoint'.alternatingCase()).to.equal('cHeCkPoInT');
+      expect('string'.alternatingCase()).to.equal('sTrInG');
+    });
+  });
+
+  describe('getMiddle', () => {
+    it('should return the character in the middle of a string', () => {
+      expect('world'.getMiddle()).to.equal('r');
+    });
+    it('should return two characters for a string with an even length', () => {
+      expect('worlds'.getMiddle()).to.equal('rl');
+    });
+  });
+
+  describe('numberWords', () => {
+    it('should return words for numbers', () => {
+      expect('805'.numberWords()).to.equal('eight zero five');
+    });
+  });
+
+  describe('isDigit', () => {
+    it('should return "true" if string is a digit', () => {
+      expect('7'.isDigit()).to.be.true;
+    });
+    it('should return "false" if string is not a single digit', () => {
+      expect('77'.isDigit()).to.be.false;
+    });
+    it('should be able to handle negative and positive operators', () => {
+      expect('-7'.isDigit()).to.be.true;
+      expect('+7'.isDigit()).to.be.true;
```

Comment:
###Critical
By definition, a digit is any single number between 0 and 9. Adding operators does not make them digits anymore.